### ISSUE TITLE
Custom dataset

### ISSUE THREAD

**Issue by AlexanderUzhinskiy** [sunjin-k/auto-augmentation#1]
Hello! I'd like to use your repository on my dataset. So I've change the data and the network like this:

import autoaug.autoaugment_learners as all
import torchvision
from torchvision import transforms
from torchvision.models import mobilenet_v2

TRAIN_DATA_PATH = "data/Pepper/train/"
TEST_DATA_PATH = "data/Pepper/test/"
TRANSFORM_IMG = transforms.Compose([
    transforms.Resize(256),
    transforms.ToTensor(),
    transforms.Normalize(mean=[0.485, 0.456, 0.406],
                         std=[0.229, 0.224, 0.225] )
    ])

train_dataset = torchvision.datasets.ImageFolder(root=TRAIN_DATA_PATH, transform=TRANSFORM_IMG)
val_dataset = torchvision.datasets.ImageFolder(root=TEST_DATA_PATH, transform=TRANSFORM_IMG)

child_network_architecture = mobilenet_v2(False)

search_space_hyp = {
        # number of subpolicies per policy
        'num_sub_policies' : 5,
        # number of bins for probability
        'p_bins' : 10,
        # number of bins for magnitude
        'm_bins' : 11,
        # image operations to exclude from
        'exclude_method' : []
        }

child_network_hyp = {
        'learning_rate' : 0.01,
        'max_epochs' : 5,
        'early_stop_num' : 15,
        'batch_size' : 16,
        'toy_size' : 1 # using a toy size of 1 means
                   # we use the whole dataset
        }

learner = aal.GenLearner(
                        **search_space_hyp,
                        **child_network_hyp,
                        num_offspring=4
                        )

learner.learn(
        train_dataset=train_dataset,
        test_dataset=val_dataset,
        child_network_architecture=child_network_architecture,
        iterations = 20)

But probably I'm doing thomething wrong. The output is:

[(('ShearX', 0.8, 3), ('TranslateY', 0.7, 3))]
main.train_child_network best accuracy:  tensor([0.], device='cuda:0')
main.train_child_network best accuracy:  tensor(0.1400, device='cuda:0')
main.train_child_network best accuracy:  tensor(0.1400, device='cuda:0')
main.train_child_network best accuracy:  tensor(0.1400, device='cuda:0')
main.train_child_network best accuracy:  tensor(0.1400, device='cuda:0')
[(('Brightness', 0.5, 0), ('Brightness', 0.0, 0))]
main.train_child_network best accuracy:  tensor([0.], device='cuda:0')
main.train_child_network best accuracy:  tensor(0.1400, device='cuda:0')
main.train_child_network best accuracy:  tensor(0.1400, device='cuda:0')
main.train_child_network best accuracy:  tensor(0.1400, device='cuda:0')
main.train_child_network best accuracy:  tensor(0.1400, device='cuda:0')
[(('Equalize', 0.1, None), ('Sharpness', 0.2, 1))]
main.train_child_network best accuracy:  tensor([0.], device='cuda:0')
main.train_child_network best accuracy:  tensor(0.1400, device='cuda:0')
main.train_child_network best accuracy:  tensor(0.1400, device='cuda:0')
main.train_child_network best accuracy:  tensor(0.1400, device='cuda:0')
main.train_child_network best accuracy:  tensor(0.1400, device='cuda:0')
...

The accuracy always 0.14. 

Can you please tell me what am I doing wrong and how to use custom dataset and network.

**Comment by sunjin-k** [comment]
Hello Alexander, thank you for your interest in our code. I'll actually recommend you take a look at 
https://paperswithcode.com/task/data-augmentation
for the current state-of-the-art automatic data augmentation methodologies instead of using our code, since this repository was mainly used as a practice for students to learn software engineering skills. Also all of our team is busy with our final dissertation which is due next Wednesday. 

To give some quick ideas: try to check your model is actually learning and not randomly guessing. Perhaps the dataset is very small and the difference in accuracy is negligible?

Best of luck with your project!

**Comment by AlexanderUzhinskiy** [comment]
Hello!

Thank you for your recomendation, but it turns rather dificult to use those repositories...

The problem probably was in the data. I decrised learning rate, increased batch size and number of epochs, and also used Adam instead off SGD.  Now I can see changes in accurasy.
